trinity/protocol/common/handlers.py

```diff
@@ -17,24 +17,28 @@
 
 
 class BaseExchangeHandler:
+    _exchange_managers: Set[ExchangeManager[Any, Any, Any]]
+
     @property
     @abstractmethod
     def _exchanges(self) -> Dict[str, Type[BaseExchange[Any, Any, Any]]]:
         pass
 
     def __init__(self, peer: BasePeer) -> None:
         self._peer = peer
+        self._exchange_managers = set()
 
         for attr, exchange_cls in self._exchanges.items():
             if hasattr(self, attr):
                 raise AttributeError(
                     "Unable to set manager on attribute `{0}` which is already "
                     "present on the class: {1}".format(attr, getattr(self, attr))
                 )
-            manager: ExchangeManager[Any, Any, Any] = ExchangeManager(self._peer, peer.cancel_token)
+            manager: ExchangeManager[Any, Any, Any]
+            manager = ExchangeManager(self._peer, exchange_cls.response_cmd_type, peer.cancel_token)
```

Comment:
IIUC the `classproperty` is being used to reduce the length of this lookup from `exchance_cls.request_class.reponse_cmd_type` to `exchange_cls.response_cmd_type`.

I'm not convinced that the `classproperty` decorator is worth this given that it required us to silence `mypy`.

Reply:
Got rid of the mypy silencer, so I kept the class property thing.